operator/cmd/syncclusterconfig/sync.go

```diff
@@ -125,3 +135,83 @@ func loadBoostrapYAML(path string) (map[string]any, error) {
 
 	return config, nil
 }
+
+// maybeMergeSuperusers merges the values found in users.txt into our superusers
+// list found in our bootstrap.yaml. This only occurs if an entry for superusers
+// actually exists in the bootstrap.yaml.
+func maybeMergeSuperusers(logger logr.Logger, clusterConfig map[string]any, path string) {
+	if path == "" {
+		// we have no path to a users.txt, so don't do anything
+		logger.Info("--users-txt not specified. Skipping superusers merge.")
+		return
+	}
+
+	superusersConfig, ok := clusterConfig[superusersEntry]
+	if !ok {
+		// we have no superusers configuration, so don't do anything
+		logger.Info("Configuration does not contain a 'superusers' entry. Skipping superusers merge.")
```

Comment:
Why syncer can not overwrite/create superusers even if there is no bootstrap user created in the first place?

Reply:
I just wanted to be a bit more in-line with what was going on before and conservative about the change. Didn't want to touch the field if we weren't going to set it already.